specification/communication/data-plane/JobRouter/preview/2021-10-20-preview2/communicationservicejobrouter.json

```diff
@@ -0,0 +1,3635 @@
+        "scoringRuleOptions": {
+          "$ref": "#/definitions/ScoringRuleOptions"
+        }
+      },
+      "x-ms-discriminator-value": "best-worker"
+    },
+    "CancelExceptionAction": {
+      "description": "An action that marks a job as cancelled",
+      "type": "object",
+      "allOf": [
+        {
+          "$ref": "#/definitions/ExceptionAction"
+        }
+      ],
+      "properties": {
+        "kind": {
+          "type": "string",
+          "readOnly": true
+        },
+        "note": {
+          "description": "(Optional) A note that will be appended to the jobs' Notes collection with th current timestamp.",
+          "maxLength": 2000,
+          "minLength": 0,
+          "type": "string"
+        },
+        "dispositionCode": {
+          "description": "(Optional) Indicates the outcome of the job, populate this field with your own custom values.",
+          "maxLength": 500,
+          "minLength": 0,
+          "type": "string"
+        }
+      },
+      "x-ms-discriminator-value": "cancel"
+    },
+    "CancelJobRequest": {
+      "description": "Request payload for deleting a job",
+      "type": "object",
+      "properties": {
+        "note": {
+          "description": "(Optional) A note that will be appended to the jobs' Notes collection with th current timestamp.",
+          "maxLength": 2000,
+          "minLength": 0,
+          "type": "string"
+        },
+        "dispositionCode": {
+          "description": "Indicates the outcome of the job, populate this field with your own custom values.\r\nIf not provided, default value of \"Cancelled\" is set.",
+          "maxLength": 500,
+          "minLength": 0,
+          "type": "string"
+        }
+      }
+    },
+    "CancelJobResponse": {
+      "type": "object"
+    },
+    "ChannelConfiguration": {
+      "required": [
+        "capacityCostPerJob"
+      ],
+      "type": "object",
+      "properties": {
+        "capacityCostPerJob": {
+          "format": "int32",
+          "description": "The amount of capacity that an instance of a job of this channel will consume of the total worker capacity.",
+          "type": "integer"
+        }
+      }
+    },
+    "ClassificationPolicy": {
+      "description": "A container for the rules that govern how jobs are classified.",
+      "type": "object",
+      "properties": {
+        "id": {
+          "description": "Unique identifier of this policy.",
+          "type": "string",
+          "readOnly": true
+        },
+        "name": {
+          "description": "Friendly name of this policy.",
+          "type": "string"
+        },
+        "fallbackQueueId": {
+          "description": "The fallback queue to select if the queue selector doesn't find a match.",
+          "type": "string"
+        },
+        "queueSelectors": {
+          "description": "The queue selectors to resolve a queue for a given job.",
+          "type": "array",
+          "items": {},
+          "x-nullable": true
```

Comment:
`Avoid the use of x-nullable.` -- This is very difficult for downstream code gen.

Reply:
Tracked using [issue](https://skype.visualstudio.com/SPOOL/_workitems/edit/2788082)

Reply:
After investigation it seems like it's beneficial for client sdks to keep `x-nullable: true`. 

When we removed x-nullable flag, the client was modified from `property = null` to property.ThrowNonNullablePropertyIsNull(). This will throw JsonException in customer applications in Debug build.

"This exception only happens in DEBUG builds of the library and would be ignored in the release build" -- https://cs.github.com/Azure/autorest.csharp/blob/6915dbe22302dd54ea7883d0c51bdb917c0adb6c/src/assets/Generator.Shared/JsonElementExtensions.cs#L95

More discussion is here: https://skype.visualstudio.com/SCC/_git/ic3_spool_ccaap/pullrequest/603633?discussionId=6011870